cg/apps/housekeeper/hk.py

```diff
@@ -662,3 +664,40 @@ def reset_retrieved_archive_data(self, files_to_reset: list[File]):
             file.archive.retrieval_task_id = None
             file.archive.retrieved_at = None
         self.commit()
+
+    @staticmethod
+    def get_files_with_tags(files: list[File], tags: list[set[str]]) -> list[File]:
+        """Return files containing specified tags."""
+        filtered_files: list[File] = []
+        for file in files:
+            file_tags: set[str] = {tag.name for tag in file.tags}
+            if any(tag.issubset(file_tags) for tag in tags):
+                filtered_files.append(file)
+        return filtered_files
+
+    @staticmethod
+    def get_files_without_excluded_tags(
+        files: list[File], excluded_tags: list[set[str]]
+    ) -> list[File]:
+        """Return files without specified tags."""
+        filtered_files: list[File] = []
+        for file in files:
+            file_tags: set[str] = {tag.name for tag in file.tags}
+            if all(not excluded_tag.issubset(file_tags) for excluded_tag in excluded_tags):
+                filtered_files.append(file)
+        return filtered_files
+
+    def get_files_from_latest_version_with_tags(
+        self, bundle_name: str, tags: list[set[str]], excluded_tags: list[set[str]] | None = None
+    ) -> list[File]:
+        """
+        Return files from the latest version of a bundle matching provided tags. Files containing
+        any tag sets specified in the excluded_tags list will be excluded from the output.
+        """
+        files: list[File] = self.get_files_from_latest_version(bundle_name=bundle_name)
```

Comment:
Can you not just pass the tags in this call and not need the middle step with "self.get_files_with_tags"?

Reply:
No, I cannot. The `get_files_from_latest_version ` function calls `FILTER_FILES_BY_TAGS`, which returns the files that match the exact tag list. Here, we want all the files that contain the provided tags (i.e. provided tags are subset of file tags)

Reply:
Maybe it would it be clearer if we rename `get_files_with_tags` to `get_files_containing_tags`... 🤔

Reply:
I think the unclearest part of this is that filtering on tags requires an exact match of the sets 😅 But oh well, then that is reasonable.

Reply:
But maybe `get_files_containing_tags` might make the distinction clearer

Reply:
Yes, I think it's clearer, I've changed it. Thanks for the review @islean!